From 4f94a1f5f1126be74b9f5cea168a378fcca3a6a6 Mon Sep 17 00:00:00 2001
From: Carl Schwan <carl@carlschwan.eu>
Date: Thu, 7 Nov 2024 17:07:47 +0000
Subject: [PATCH] Fix desktop file for flathub linting

(cherry picked from commit d58ea434e74cb73adfffeac0d8ddbe4ec07a9ff8)

Co-authored-by: Carl Schwan <carl@carlschwan.eu>
---
 src/CMakeLists.txt                                         | 7 +++++++
 src/launcher/CMakeLists.txt                                | 1 -
 .../org.kde.merkuro.desktop => org.kde.merkuro.desktop.in} | 3 ++-
 3 files changed, 9 insertions(+), 2 deletions(-)
 rename src/{launcher/org.kde.merkuro.desktop => org.kde.merkuro.desktop.in} (86%)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0351d5bf0..3b0df9a96 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -9,6 +9,13 @@ add_subdirectory(calendar)
 add_subdirectory(components)
 if (MERKURO_FLATPAK)
     add_subdirectory(launcher)
+    set(DESKTOP_EXEC "merkurolauncher")
+    set(NO_DISPLAY "false")
+else()
+    set(DESKTOP_EXEC "merkuro-calendar")
+    set(NO_DISPLAY "true")
 endif()
 
+configure_file(org.kde.merkuro.desktop.in ${CMAKE_CURRENT_BINARY_DIR}/org.kde.merkuro.desktop)
+install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/org.kde.merkuro.desktop DESTINATION ${KDE_INSTALL_APPDIR})
 install(FILES org.kde.merkuro.metainfo.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
diff --git a/src/launcher/CMakeLists.txt b/src/launcher/CMakeLists.txt
index 0f25aded1..c2a3dfc7a 100644
--- a/src/launcher/CMakeLists.txt
+++ b/src/launcher/CMakeLists.txt
@@ -29,4 +29,3 @@ target_link_libraries(merkurolauncher PRIVATE
 endif()
 
 install(TARGETS merkurolauncher ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
-install(PROGRAMS org.kde.merkuro.desktop DESTINATION ${KDE_INSTALL_APPDIR})
diff --git a/src/launcher/org.kde.merkuro.desktop b/src/org.kde.merkuro.desktop.in
similarity index 86%
rename from src/launcher/org.kde.merkuro.desktop
rename to src/org.kde.merkuro.desktop.in
index a6fe18dfa..a3a8db6ed 100644
--- a/src/launcher/org.kde.merkuro.desktop
+++ b/src/org.kde.merkuro.desktop.in
@@ -3,7 +3,8 @@
 [Desktop Entry]
 Type=Application
 Name=Merkuro
-Exec=merkurolauncher
+Exec=@DESKTOP_EXEC@
+NoDisplay=@NO_DISPLAY@
 GenericName=Personal Information Manager
 Icon=org.kde.merkuro.calendar
 X-KDE-StartupNotify=true
-- 
GitLab

